ddl/syncer.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb/terror"
+	goctx "golang.org/x/net/context"
+)
+
+const (
+	ddlSelfSchemaVersion   = "tidb/ddl/self_schem_version"
+	ddlLatestSchemaVersion = "tidb/ddl/latest_schem_version"
```

Comment:
I think it's OK, but I'd like to change "schem" to "schema"
'ddlAllSchemaVersions' and 'ddlGlobalSchemaVersion' is more confusing for me.